src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.h

```diff
@@ -28,50 +28,50 @@
 class TargetedMovementGeneratorBase
 {
     public:
-        TargetedMovementGeneratorBase(Unit* target) { i_target.link(target, this); }
-        void stopFollowing() { }
+        TargetedMovementGeneratorBase(Unit* target) { m_target.link(target, this); }
+        void StopFollowing() { }
     protected:
-        FollowerReference i_target;
+        FollowerReference m_target;
 };
 
 template<class T, typename D>
 class TargetedMovementGeneratorMedium : public MovementGeneratorMedium< T, D >, public TargetedMovementGeneratorBase
 {
-    protected:
-        TargetedMovementGeneratorMedium(Unit* target, float offset, float angle) :
-            TargetedMovementGeneratorBase(target), i_path(NULL),
-            i_recheckDistance(0), i_offset(offset), i_angle(angle),
-            i_recalculateTravel(false), i_targetReached(false)
-        {
-        }
-        ~TargetedMovementGeneratorMedium() { delete i_path; }
-
     public:
+        TargetedMovementGeneratorMedium(Unit* target, float offset, float angle) : TargetedMovementGeneratorBase(target),
+            m_path(nullptr), m_recheckDistance(0), m_offset(offset), m_angle(angle), m_recalculateTravel(false), m_targetReached(false), m_interrupt(false) { }
+
+        ~TargetedMovementGeneratorMedium();
+
         bool DoUpdate(T*, uint32);
-        Unit* GetTarget() const { return i_target.getTarget(); }
+        Unit* GetTarget() const { return m_target.getTarget(); }
+        void DoInterrupt(T*);
 
-        void unitSpeedChanged() override { i_recalculateTravel = true; }
-        bool IsReachable() const { return (i_path) ? (i_path->GetPathType() & PATHFIND_NORMAL) : true; }
+        void UnitSpeedChanged() override { m_recalculateTravel = true; }
+        void UnitMovementInterrupted(Unit* unit) override { DoInterrupt(static_cast<T*>(unit)); }
+
+        virtual void ClearUnitStateMove(T*) { }
+        virtual void AddUnitStateMove(T*) { }
+
+        bool IsReachable() const { return (m_path) ? (m_path->GetPathType() & PATHFIND_NORMAL) : true; }
     protected:
-        void _setTargetLocation(T* owner, bool updateDestination);
-
-        PathGenerator* i_path;
```

Comment:
m_ and i_ prefixes are not hungarian notation, hungarians would appending type to name (for example szStringfield, uint32 uiIntfield)